File: Go/bubblesort.go
package main

import "fmt"

func bubblesort(array []int) {
	for i := 0; i < len(array) - 1; i++ {
		for j := 1; j < len(array) - i; j++ {
			if array[j - 1] > array[j] {
				array[j - 1], array[j] = array[j], array[j - 1]
			}
		}
	}
}

func sort(array []int) {
	bubblesort(array)
}

func main() {
	array := []int{2, 0, 7, 3, 4, 6, 5, 8, 1, 9}
	fmt.Println("Source:", array);

	sort(array)
	fmt.Println("Result:", array);
}

File: C/bubblesort.c
#include <stdio.h>

int bubblesort(int* array, int length)
{
	int i, j, tmp;

	for (i = 0; i < length - 1; i++) {
		for (j = 1; j < length - i; j++) {
			if (array[j - 1] > array[j]) {
				tmp = array[j - 1];
				array[j - 1] = array[j];
				array[j] = tmp;
			}
		}
	}
}

int sort(int* array, int length)
{
	bubblesort(array, length);
}

int main(int argc, char* argv[])
{
	int array[] = {2, 0, 7, 3, 4, 6, 5, 8, 1, 9};
	int length = sizeof(array) / sizeof(int);

	sort(array, length);

#if 0
	int i = 0;
	for (i; i < length; i++)
		printf("%d ", array[i]);
	puts("");
#endif
	return 0;
}